pcsx2-qt/QtUtils.cpp

```diff
@@ -139,6 +140,90 @@ namespace QtUtils
 		ResizeColumnsForView(view, widths);
 	}
 
+
+	static int DPRScale(int size, qreal dpr)
+	{
+		return static_cast<int>(static_cast<qreal>(size) * dpr);
+	}
+
+	static int DPRUnscale(int size, qreal dpr)
+	{
+		return static_cast<int>(static_cast<qreal>(size) / dpr);
+	}
+
+	void resizeAndPadPixmap(QPixmap* pm, int expected_width, int expected_height, qreal dpr, ScalingMode aspect_ratio, const float opacity)
+	{
+		const int dpr_expected_width = DPRScale(expected_width, dpr);
+		const int dpr_expected_height = DPRScale(expected_height, dpr);
+		if (pm->width() == dpr_expected_width && pm->height() == dpr_expected_height)
+			return;
+
+		switch (aspect_ratio)
+		{
+			case ScalingMode::Fit:
+			{
+				*pm = pm->scaledToHeight(dpr_expected_height, Qt::SmoothTransformation);
```

Comment:
This won't correctly fit the image if the image is wider then the window for a given height